src/flow/src/expr/func.rs

```diff
@@ -61,14 +68,51 @@ impl UnmaterializableFunc {
                 output: ConcreteDataType::string_datatype(),
                 generic_fn: GenericFn::CurrentSchema,
             },
+            Self::TumbleWindow { .. } => Signature {
+                input: smallvec![ConcreteDataType::timestamp_millisecond_datatype()],
+                output: ConcreteDataType::timestamp_millisecond_datatype(),
+                generic_fn: GenericFn::TumbleWindow,
+            },
         }
     }
 
     /// Create a UnmaterializableFunc from a string of the function name
-    pub fn from_str(name: &str) -> Result<Self, Error> {
-        match name {
+    pub fn from_str_args(name: &str, args: Vec<TypedExpr>) -> Result<Self, Error> {
+        match name.to_lowercase().as_str() {
             "now" => Ok(Self::Now),
             "current_schema" => Ok(Self::CurrentSchema),
+            "tumble" => {
+                let ts = args.first().context(InvalidQuerySnafu {
+                    reason: "Tumble window function requires a timestamp argument",
+                })?;
+                let window_size = args
+                    .get(1)
+                    .and_then(|expr| expr.expr.as_literal())
+                    .context(InvalidQuerySnafu {
+                        reason: "Tumble window function requires a window size argument"
+                    })?.as_string()// TODO(discord9): since df to substrait convertor does not support interval type yet, we need to take a string and cast it to interval instead
```

Comment:
```suggestion
                    })?.as_string() // TODO(discord9): since df to substrait convertor does not support interval type yet, we need to take a string and cast it to interval instead
```